Weird includes when installing salt using `USE_STATIC_REQUIREMENTS=1`

Installing salt using `USE_STATIC_REQUIREMENTS=1`, which selects a different set of requirements, fails.

```
ubuntu@ip-10-1-102-50:~$ salt/bin/python3 -m pip install downloaded-requirements/cffi-1.14.6.tar.gz
Processing ./downloaded-requirements/cffi-1.14.6.tar.gz
  Preparing metadata (setup.py) ... done
Requirement already satisfied: pycparser in ./salt/lib/python3.10/site-packages (from cffi==1.14.6) (2.21)
Building wheels for collected packages: cffi
  Building wheel for cffi (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [38 lines of output]
      /home/ubuntu/salt/lib/python3.10/site-packages/setuptools/config/setupcfg.py:463: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
        warnings.warn(msg, warning_class)
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-310
      creating build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/recompiler.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/vengine_cpy.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/backend_ctypes.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/setuptools_ext.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/error.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/pkgconfig.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/api.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/commontypes.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/model.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/cffi_opcode.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/vengine_gen.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/ffiplatform.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/verifier.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/__init__.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/cparser.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/lock.py -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/_cffi_include.h -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/parse_c_type.h -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/_embedding.h -> build/lib.linux-x86_64-cpython-310/cffi
      copying cffi/_cffi_errors.h -> build/lib.linux-x86_64-cpython-310/cffi
      running build_ext
      building '_cffi_backend' extension
      creating build/temp.linux-x86_64-cpython-310
      creating build/temp.linux-x86_64-cpython-310/c
      /home/ubuntu/.local/relenv/toolchain/x86_64-linux-gnu/bin/x86_64-linux-gnu-gcc -no-pie -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -O3 -Wall -L/home/ubuntu/.local/lib -L/home/ubuntu/.local/relenv/
toolchain/x86_64-linux-gnu/x86_64-linux-gnu/sysroot/lib -I/home/ubuntu/.local/include -I/home/ubuntu/.local/include/readline -I/home/ubuntu/.local/include/ncursesw -I/home/ubuntu/.local/relenv/toolchain/x86_64-lin
ux-gnu/x86_64-linux-gnu/sysroot/usr/include -L/home/ubuntu/.local/lib -L/home/ubuntu/.local/relenv/toolchain/x86_64-linux-gnu/x86_64-linux-gnu/sysroot/lib -I/home/ubuntu/.local/include -I/home/ubuntu/.local/includ
e/readline -I/home/ubuntu/.local/include/ncursesw -I/home/ubuntu/.local/relenv/toolchain/x86_64-linux-gnu/x86_64-linux-gnu/sysroot/usr/include -fPIC -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/home/ubuntu/salt/includ
e/python3.10 -c c/_cffi_backend.c -o build/temp.linux-x86_64-cpython-310/c/_cffi_backend.o
      c/_cffi_backend.c:15:10: fatal error: ffi.h: No such file or directory
         15 | #include <ffi.h>
            |          ^~~~~~~
      compilation terminated.
      error: command '/home/ubuntu/.local/relenv/toolchain/x86_64-linux-gnu/bin/x86_64-linux-gnu-gcc' failed with exit code 1
```

--- a/relenv/build/common.py
+++ b/relenv/build/common.py
@@ -59,7 +59,19 @@ SITECUSTOMIZE = """\"\"\"
 Relenv site customize
 \"\"\"
 import site, os
+
+# Remove any paths not relative to relenv's root directory.
+path = []
+root = pathlib.Path(__file__).parent.parent.parent.parent
+for _ in sys.path:
+    try:
+        if pathlib.Path(_).relative_to(root):
+            path.append(_)
+    except ValueError:
+        continue
+sys.path = path
 site.ENABLE_USER_SITE = False
+
 try:
     import relenv.runtime
 except ImportError:
@@ -67,6 +79,54 @@ except ImportError:
         print("Unable to find relenv.runtime for bootstrap.")
 else:
     relenv.runtime.bootstrap()
+"""
+
+SYSCONFIGDATA = """
+import pathlib, sys, platform, os
+
+def build_arch():
+    machine = platform.machine()
+    return machine.lower()
+
+def get_triplet(machine=None, plat=None):
+    if not plat:
+        plat = sys.platform
+    if not machine:
+        machine = build_arch()
+    if plat == "darwin":
+        return f"{machine}-macos"
+    elif plat == "win32":
+        return f"{machine}-win"
+    elif plat == "linux":
+        return f"{machine}-linux-gnu"
+    else:
+        raise RelenvException("Unknown platform {}".format(platform))
+
+
+
+pydir = pathlib.Path(__file__).resolve().parent
+if sys.platform == "win32":
+    DEFAULT_DATA_DIR = pathlib.Path.home() / "AppData" / "Local" / "relenv"
+else:
+    DEFAULT_DATA_DIR = pathlib.Path.home() / ".local" / "relenv"
+
+if "RELENV_DATA" in os.environ:
+    DATA_DIR = pathlib.Path(os.environ["RELENV_DATA"]).resolve()
+else:
+    DATA_DIR = DEFAULT_DATA_DIR
+
+buildroot = pydir.parent.parent
+toolchain = DATA_DIR / "toolchain" / get_triplet()
+build_time_vars = {}
+for key in _build_time_vars:
+    val = _build_time_vars[key]
+    orig = val
+    if isinstance(val, str):
+        val = val.format(
+            BUILDROOT=buildroot,
+            TOOLCHAIN=toolchain,
+        )
+    build_time_vars[key] = val
 """
 
 
@@ -959,8 +1019,25 @@ def install_sysdata(mod, destfile, buildroot, toolchain):
         f.write(
             "# system configuration generated and used by" " the relenv at runtime\n"
         )
-        f.write("build_time_vars = ")
+        f.write("_build_time_vars = ")
         pprint.pprint(data, stream=f)
+        f.write(SYSCONFIGDATA)
+
+
+def find_sysconfigdata(pymodules):
+    """
+    Find sysconfigdata directory for python installation
+
+    :param pymodules: Path to python modules (e.g. lib/python3.10)
+    :type pymodules: str
+
+    :return: The name of the sysconig data module
+    :rtype: str
+    """
+    for root, dirs, files in os.walk(pymodules):
+        for file in files:
+            if file.find("sysconfigdata") > -1 and file.endswith(".py"):
+                return file[:-3]
 
 
 def finalize(env, dirs, logfp):
@@ -988,12 +1065,6 @@ def finalize(env, dirs, logfp):
 
     pymodules = libdir / find_pythonlib(libdir)
 
-    def find_sysconfigdata(pymodules):
-        for root, dirs, files in os.walk(pymodules):
-            for file in files:
-                if file.find("sysconfigdata") > -1 and file.endswith(".py"):
-                    return file[:-3]
-
     cwd = os.getcwd()
     modname = find_sysconfigdata(pymodules)
     path = sys.path
@@ -1003,7 +1074,8 @@ def finalize(env, dirs, logfp):
     finally:
         os.chdir(cwd)
         sys.path = path
-    dest = pymodules / "site-packages" / "relenv-sysconfigdata.py"
+
+    dest = pymodules / f"{modname}.py"
     install_sysdata(mod, dest, dirs.prefix, dirs.toolchain)
 
     # Lay down site customize

--- a/relenv/build/common.py
+++ b/relenv/build/common.py
@@ -58,7 +58,7 @@ NODOWLOAD = False
 SITECUSTOMIZE = """\"\"\"
 Relenv site customize
 \"\"\"
-import site, os
+import os, pathlib, site, sys
 
 # Remove any paths not relative to relenv's root directory.
 path = []

--- a/relenv/common.py
+++ b/relenv/common.py
@@ -14,7 +14,7 @@ import urllib.error
 import urllib.request
 
 # relenv package version
-__version__ = "0.3.0"
+__version__ = "0.4.0"
 
 MODULE_DIR = pathlib.Path(__file__).resolve().parent
 

--- a/relenv/runtime.py
+++ b/relenv/runtime.py
@@ -97,7 +97,6 @@ class RelenvImporter:
     loading_sysconfig_data = False
     loading_sysconfig = False
 
-    build_time_vars = None
     sysconfigdata = "_sysconfigdata__linux_x86_64-linux-gnu"
 
     def find_module(self, module_name, package_path=None):
@@ -122,12 +121,6 @@ class RelenvImporter:
                 return None
             debug(f"RelenvImporter - match {module_name}")
             self.loading_pip_scripts = True
-            return self
-        elif module_name == self.sysconfigdata:
-            if self.loading_sysconfig_data:
-                return None
-            debug(f"RelenvImporter - match {module_name}")
-            self.loading_sysconfig_data = True
             return self
         return None
 
@@ -158,57 +151,8 @@ class RelenvImporter:
             mod = importlib.import_module(name)
             mod.ScriptMaker._build_shebang = _build_shebang
             self.loading_pip_scripts = False
-        elif name == self.sysconfigdata:
-            debug(f"RelenvImporter - load_module {name}")
-            mod = importlib.import_module(name)
-            try:
-                relmod = importlib.import_module("relenv-sysconfigdata")
-            except ImportError:
-                debug("Unable to import relenv-sysconfigdata")
-                return mod
-            debug(f"RelenvImporter - MODULE_DIR {MODULE_DIR}")
-            buildroot = MODULE_DIR.parent.parent.parent.parent
-            toolchain = DATA_DIR / "toolchain" / get_triplet()
-            build_time_vars = {}
-            for key in relmod.build_time_vars:
-                val = relmod.build_time_vars[key]
-                orig = val
-                if isinstance(val, str):
-                    val = val.format(
-                        BUILDROOT=buildroot,
-                        TOOLCHAIN=toolchain,
-                    )
-                build_time_vars[key] = val
-                # self.build_time_vars.build_time_vars = build_time_vars
-                mod.build_time_vars = build_time_vars
-            self.loading_sysconfig_data = False
-            return self
         sys.modules[name] = mod
         return mod
-
-
-class BuildTimeVars(collections.abc.Mapping):
-    """
-    A dictionary-like object holding build-time variables.
-    """
-
-    # This is getting set in RelenvImporter
-    _build_time_vars = {}
-
-    def __getitem__(self, key, *args, **kwargs):
-        debug(f"BuildTimeVars - getitem {name}")
-        val = self._build_time_vars.__getitem__(key, *args, **kwargs)
-        return val.format(
-            BUILDROOT=self.buildroot,
-            TOOLCHAIN=self.toolchain,
-        )
-        return val
-
-    def __iter__(self):
-        return iter(self._build_time_vars)
-
-    def __len__(self):
-        return len(self._build_time_vars)
 
 
 def bootstrap():
@@ -253,7 +197,5 @@ def bootstrap():
         path = pathlib.Path(_.strip().strip('"'))
         os.environ["SSL_CERT_DIR"] = str(path / "certs")
         os.environ["SSL_CERT_FILE"] = str(path / "cert.pem")
-    build_time_vars = BuildTimeVars()
     importer = RelenvImporter()
-    importer.build_time_vars = build_time_vars
     sys.meta_path = [importer] + sys.meta_path

--- a/relenv/runtime.py
+++ b/relenv/runtime.py
@@ -166,6 +166,7 @@ class RelenvImporter:
             except ImportError:
                 debug("Unable to import relenv-sysconfigdata")
                 return mod
+            debug(f"RelenvImporter - MODULE_DIR {MODULE_DIR}")
             buildroot = MODULE_DIR.parent.parent.parent.parent
             toolchain = DATA_DIR / "toolchain" / get_triplet()
             build_time_vars = {}

--- a/tests/test_verify_build.py
+++ b/tests/test_verify_build.py
@@ -133,7 +133,7 @@ def test_pip_install_salt_w_static_requirements(pipexec, build):
     env = os.environ.copy()
     env["RELENV_DEBUG"] = "yes"
     env["USE_STATIC_REQUIREMENTS"] = "1"
-    p = subprocess.run(["git", "clone", "git@github.com:saltstack/salt.git"])
+    p = subprocess.run(["git", "clone", "https://github.com/saltstack/salt.git"])
 
     p = subprocess.run([str(pipexec), "install", "./salt", "--no-cache"], env=env)
     assert p.returncode == 0, f"Failed to pip install ./salt"

--- a/tests/test_verify_build.py
+++ b/tests/test_verify_build.py
@@ -129,6 +129,26 @@ def test_pip_install_salt(pipexec, build):
             script = pathlib.Path(build) / "bin" / _
         assert script.exists()
 
+def test_pip_install_salt_w_static_requirements(pipexec, build):
+    env = os.environ.copy()
+    env["RELENV_DEBUG"] = "yes"
+    env["USE_STATIC_REQUIREMENTS"] = "1"
+    p = subprocess.run(["git", "clone", "git@github.com:saltstack/salt.git"])
+
+    p = subprocess.run([str(pipexec), "install", "./salt", "--no-cache"], env=env)
+    assert p.returncode == 0, f"Failed to pip install ./salt"
+
+    names = ["salt", "salt-call", "salt-master", "salt-minion"]
+    if sys.platform == "win32":
+        names = ["salt-call.exe", "salt-minion.exe"]
+
+    for _ in names:
+        if sys.platform == "win32":
+            script = pathlib.Path(build) / "Scripts" / _
+        else:
+            script = pathlib.Path(build) / "bin" / _
+        assert script.exists()
+
 
 def test_pip_install_cryptography(pipexec):
     packages = [

